packages/common/src/types/warehouse.ts

```diff
@@ -100,7 +109,28 @@ export type ApiWarehouseCatalog = {
     results: WarehouseCatalog;
 };
 
+export type ApiWarehouseTablesCatalog = {
+    status: 'ok';
+    results: WarehouseTablesCatalog;
+};
+
 export type ApiWarehouseTableFields = {
     status: 'ok';
     results: WarehouseTableSchema;
 };
+
+export enum PartitionType {
+    DATE = 'DATE',
+    RANGE = 'RANGE',
+}
+export type PartitionColumnDate = {
```

Comment:
These types are an extension of what bigquery returns on their metadata (+ partitionType for easy type checking)